src/google/protobuf/descriptor.proto

```diff
@@ -481,7 +490,11 @@ message FieldOptions {
   // For Google-internal migration only. Do not use.
   optional bool weak = 10 [default=false];
 
-
+  // Provides the ability to override the name of the property
+  // generated for this field. This is applied to all properties
+  // and methods to do with this field, including HasFoo, FooCount,
+  // FooList etc.
+  optional string csharp_property_name = 11;
```

Comment:
Why is this option needed? Can we move less common options into an extension based option defined in the csharp runtime?